torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -210,64 +210,73 @@ def resize_image(
         # error if other interpolation modes are used. This is documented.
         antialias = False
 
-    shape = image.shape
-    numel = image.numel()
-    num_channels, old_height, old_width = shape[-3:]
+    old_height, old_width = image.shape[-2:]
     new_height, new_width = _compute_resized_output_size((old_height, old_width), size=size, max_size=max_size)
 
     if (new_height, new_width) == (old_height, old_width):
         return image
-    elif numel > 0:
-        image = image.reshape(-1, num_channels, old_height, old_width)
 
-        dtype = image.dtype
-        acceptable_dtypes = [torch.float32, torch.float64]
-        if interpolation == InterpolationMode.NEAREST or interpolation == InterpolationMode.NEAREST_EXACT:
-            # uint8 dtype can be included for cpu and cuda input if nearest mode
+    if image.ndim > 2:
+        num_channels = image.shape[-3]
+        output_shape = image.shape[:-3] + (num_channels, new_height, new_width)
+    else:
+        num_channels = 1
+        output_shape = [new_height, new_width]  # type: ignore[assignment]
```

Comment:
Need to silence `mypy` here, because torchscript is doing something stupid here.

```python
image.shape[:-3] + (num_channels, new_height, new_width)
```

is a tuple, but internally this is represented as list. Meaning, I need to match that in the `else` branch as well. But `mypy` is picking up on that and complains that we now have a list or a tuple in eager mode depending on the branch.

There are two more patterns of this.